base/data/gm4/tags/blocks/foliage.json

```diff
@@ -0,0 +1,21 @@
+{
+  "values": [
+    "#minecraft:flowers",
```

Comment:
Missing weeping_vines_plant and twisting_vines_plant

Should we add #minecraft:crops? nether_wart? bamboo_sapling?

Reply:
I don't fully understand what foliage will be used fore hence my previous comment:
> `foliage` seems to be a bit of an odd one; I've added some things based on the assumption that `foliage` is meant to be "plants that generate naturally outside of structures". If that definition matches the intended purpose of `foliage` we should be clear to merge this.
I think that this tag is intended for naturally generating foliage to act as a way to decide which blocks can be replaced without a significant loss, i.e. naturally generating foliage that can be washed away by water; Which means it's also lacking the chorus stems.

If we were to add `#minecraft:crops` we should also add `#minecraft:saplings`, `sugar_cane`, cocoa beans, `nether_wart` and `bamboo_sapling`.

Reply:
Looking through the PR again, it seems the `gm4:foliage` tag is used for desire lines and podzol rooting soil. It's likely those two data packs had similar block tags. We should probably just look at those original block tags and compare them. 

desire lines:
     "minecraft:grass",		
     "minecraft:fern",		
     "minecraft:dead_bush",		
     "minecraft:sunflower",		
     "minecraft:lilac",		
     "minecraft:rose_bush",		
     "minecraft:peony",		
     "minecraft:tall_grass",		
     "minecraft:large_fern",		
     "minecraft:dandelion",		
     "minecraft:poppy",		
     "minecraft:blue_orchid",		
     "minecraft:allium",		
     "minecraft:azure_bluet",		
     "minecraft:red_tulip",		
     "minecraft:orange_tulip",		
     "minecraft:white_tulip",		
     "minecraft:pink_tulip",		
     "minecraft:oxeye_daisy",		
     "minecraft:red_mushroom",		
     "minecraft:brown_mushroom"

podzol rooting soil:
     "minecraft:air",		
     "minecraft:cave_air",		
     "minecraft:void_air",		
     "minecraft:grass",		
     "#minecraft:flowers",		
     "minecraft:fern",		
     "minecraft:brown_mushroom",		
     "minecraft:red_mushroom",		
     "minecraft:dead_bush",		
     "minecraft:large_fern",		
     "minecraft:tall_grass",		
     "minecraft:vine"

Reply:
Hmm actually, it seems that this PR would break podzol rooting soil, as the data pack checks for air along with foliage... 

I think the current version of foliage is good, but Podzol Rooting Soil needs another block tag containing `#gm4:air` and `#gm4:foliage` to work properly. (Desire Lines would check for `#gm4:foliage`)

Reply:
That does indeed confirm my theory of "naturally generating foliage that can be washed away by water". I'd explicitly exclude chorus stems and cocoa beans, as they actually have some collision.
The pre-existing tags in those two modules might not be complete, so they should only act as a subset for the new `foliage` tag.